modules/math/shared/src/main/scala/gsp/math/Parallax.scala

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gsp.math
+
+import cats._
+import coulomb._
+import coulomb.cats.implicits._
+import gsp.math.optics._
+import gsp.math.units._
+import spire.math.Rational
+import spire.std.long._
+
+/**
+  * Parallax stored as microarcseconds
+  * Normally parallax is expressed in milliarcseconds but simbad reports them
+  * with a higher precision thus using microarcseconds gives us enough space
+  * Absolute allowed parallax values need to be less or equal thán 180 degrees
+  */
+sealed abstract case class Parallax protected (μas: Quantity[Long, MicroArcSecond]) {
+  val mas: Quantity[Rational, MilliArcSecond] = μas.to[Rational, MilliArcSecond]
+
+  override def toString: String =
+    s"Parallax(${μas.show})"
+}
+
+object Parallax extends ParallaxOptics {
+  val MaxValue: Parallax = apply(648000000000L)
+  val MinValue: Parallax = apply(-648000000000L)
+
+  /**
+    * The `No parallax`
+    * @group Constructors
+    */
+  val Zero: Parallax = apply(0L)
+
+  // Internal unbounded constructor
+  private def apply(μas: Long): Parallax =
+    apply(μas.withUnit[MicroArcSecond])
+
+  // Internal unbounded constructor
+  private def apply(μas: Quantity[Long, MicroArcSecond]): Parallax =
+    new Parallax(μas) {}
+
+  /** @group Typeclass Instances */
+  implicit val orderParallax: Order[Parallax] =
+    Order.by(_.μas)
+
+  /** @group Typeclass Instances */
+  implicit val monoidParallax: Monoid[Parallax] =
+    Monoid.instance(Zero, (a, b) => apply(a.μas + b.μas))
+
+  /**
+    * Construct a new Parallax of the given magnitude in integral microarcseconds, modulo 180°. Exact.
+    * @group Constructors
+    */
+  def fromMicroarcseconds(μas: Long): Parallax = {
+    val µasPer180 = 180L * 60L * 60L * 1000L * 1000L
+    val µasʹ      = μas % µasPer180
+    apply(µasʹ)
```

Comment:
I feel silly for questioning the parallax math so much because the closest star has a parallax of just 0.7687 arcseconds.   Nothing should have a parallax of over 1 arcsecond. I wonder if we should set the limit to +/- 1 arcsecond?

At any rate, 181 degrees seems like it should be equivalent to 179 (-179 perhaps) rather than 1 degree. Maybe the thing to do is to use an `Angle` to get the equivalent signed μas? For example: `apply(Angle.signedMicroarcseconds.normalize(µas))`